torch_xla/csrc/device.h

```diff
@@ -2,38 +2,65 @@
 #define XLA_TORCH_XLA_CSRC_DEVICE_H_
 
 #include <torch/csrc/lazy/backend/backend_device.h>
-#include <torch/csrc/lazy/core/hash.h>
-#include <torch/csrc/lazy/core/util.h>
 
-#include <iostream>
 #include <string>
+#include <string_view>
 
-#include "torch_xla/csrc/runtime/util.h"
+#include "absl/status/statusor.h"
 
 namespace torch_xla {
 
+// Convenient macro for applying another macro to all native device types.
+//
+// Add new device type
+// ===================
+//
+// Add a new line to the macro below:
+//
+//     _(<DEVICE>, <INDEX>)
+//
+// Where <DEVICE> is the enum of the given device, and <INDEX> is the
+// previous number plus 1.
+//
+#define XLA_FOR_ALL_NATIVE_DEVICE_TYPES_(_) \
```

Comment:
Actually I'd remove this macro as it's only used once in the prod code. The uses in tests should be removed as test code needs to be obviously correct and straightforward. We don't want macros in the way of debugging/understanding tests.